lib/Sema/CSSimplify.cpp

```diff
@@ -8391,6 +8428,43 @@ ConstraintSystem::simplifyBindTupleOfFunctionParamsConstraint(
   return SolutionKind::Solved;
 }
 
+ConstraintSystem::SolutionKind
+ConstraintSystem::simplifyPackElementOfConstraint(Type first, Type second,
+                                                  TypeMatchOptions flags,
+                                                  ConstraintLocatorBuilder locator) {
+  auto elementType = simplifyType(first, flags);
+  auto *loc = getConstraintLocator(locator);
```

Comment:
Nit: Might be better to materialize locator only when we are avoid to add unsolved constraint since the other case takes a builder anyway...